app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java

```diff
@@ -130,15 +137,25 @@ protected void initializeWorld() {
         mBrowserWidget = new BrowserWidget(this, currentSession);
         mPermissionDelegate.setParentWidgetHandle(mBrowserWidget.getHandle());
 
+        // Settings Widget
+        mSettingsWidget = new SettingsWidget(VRBrowserActivity.this);
+        mSettingsWidget.hide();
+
         // Create Browser navigation widget
-        NavigationBarWidget navigationBar = new NavigationBarWidget(this);
-        navigationBar.setBrowserWidget(mBrowserWidget);
+        mNavigationBar = new NavigationBarWidget(this);
+        mNavigationBar.setBrowserWidget(mBrowserWidget);
+        mNavigationBar.getPlacement().parentHandle = mBrowserWidget.getHandle();
 
         // Create keyboard widget
         mKeyboard = new KeyboardWidget(this);
-        mKeyboard.setBrowserWidget(mBrowserWidget);
+        mKeyboard.getPlacement().parentHandle = mBrowserWidget.getHandle();
```

Comment:
Why did you change this, rebase problem? the change will break some keyboard functionality

Reply:
Yes, I think it was a rebase issue. Fixed